### PR TITLE
wasmer: initial integration

### PR DESCRIPTION
**Pull request by manunio**
Hi, [ Wasmer](https://github.com/wasmerio/wasmer) is a fast and secure [WebAssembly](https://webassembly.org/) runtime that enables super lightweight containers to run anywhere: from Desktop to the Cloud, Edge and IoT devices.
- It has close to 2 million downloads as per [crates.io](https://crates.io/crates/wasmer).
- It is being used by projects like:
  - [swc](https://github.com/swc-project/swc)
  - [zellij](https://github.com/zellij-org/zellij)
  - [youki](https://github.com/containers/youki)
  - [lunatic](https://github.com/lunatic-solutions/lunatic)
  - [mun](https://github.com/mun-lang/mun)
  - [locutus](https://github.com/freenet/locutus)

### CODE REVIEW

**Comment by jonathanmetzman** [comment]
> Hi, [ Wasmer](https://github.com/wasmerio/wasmer) is a fast and secure [WebAssembly](https://webassembly.org/) runtime that enables super lightweight containers to run anywhere: from Desktop to the Cloud, Edge and IoT devices.
> 
> * It has close to 2 million downloads as per [crates.io](https://crates.io/crates/wasmer).
> * It is being used by projects like:
>   
>   * [swc](https://github.com/swc-project/swc)
>   * [zellij](https://github.com/zellij-org/zellij)
>   * [youki](https://github.com/containers/youki)
>   * [lunatic](https://github.com/lunatic-solutions/lunatic)
>   * [mun](https://github.com/mun-lang/mun)
>   * [locutus](https://github.com/freenet/locutus)

I don't know any of these projects unfortunately. Is wasmer used or developed by any big companies?

**Comment by syrusakbary** [comment]
> Is wasmer used or developed by any big companies?

Yes. There are multiple companies. Google, for example, is one of those (via the [dart-wasm](https://github.com/dart-lang/wasm) integration). Apple, with swift is another. Kubernetes with [crun](https://github.com/containers/crun), but there are many more companies and projects from big to medium to small.

Is worth to mention that Wasmer is currently [the most popular standalone WebAssembly runtime](https://ossinsight.io/analyze/wasmerio/wasmer#overview), and I believe is running in more production setups than any other Wasm runtime already integrated in the oss-fuzz project.
Currently, oss-fuzz is only fuzzing one wasm-to-asm compiler (Cranelift, via wasmtime), but with Wasmer it will open the fuzzing to more backend compilers (LLVM and Singlepass, both used in production environments by many companies).

Integrating Wasmer in the oss-fuzz project will incredibly help the WebAssembly ecosystem as a whole, since many projects already depend on it and it will help improve resilience of the ecosystem. Thanks a lot for both considering Wasmer and integrating into the great oss-fuzz project!

**Comment by jonathanmetzman** [comment]
> > Is wasmer used or developed by any big companies?
> 
> Yes. There are multiple companies. Google, for example, is one of those (via the [dart-wasm](https://github.com/dart-lang/wasm) integration). Apple, with swift is another. Kubernetes with [crun](https://github.com/containers/crun), but there are many more companies and projects from big to medium to small.
> 
> Is worth to mention that Wasmer is currently [the most popular standalone WebAssembly runtime](https://ossinsight.io/analyze/wasmerio/wasmer#overview), and I believe is running in more production setups than any other Wasm runtime already integrated in the oss-fuzz project. Currently, oss-fuzz is only fuzzing one wasm-to-asm compiler (Cranelift, via wasmtime), but with Wasmer it will open the fuzzing to more backend compilers (LLVM and Singlepass, both used in production environments by many companies).
> 
> Integrating Wasmer in the oss-fuzz project will incredibly help the WebAssembly ecosystem as a whole, since many projects already depend on it and it will help improve resilience of the ecosystem. Thanks a lot for both considering Wasmer and integrating into the great oss-fuzz project!

I think this sounds good enough for me.
I wasn't really doubting it, was more asking a question.
Thanks!

File: projects/wasmer/deterministic.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/*
```
Comment (jonathanmetzman) [inline projects/wasmer/deterministic.rs:1]: Consider moving this stuff upstream.
I noticed you never apply for ideal integrations, you know you can make more money by following the steps here:
https://google.github.io/oss-fuzz/advanced-topics/ideal-integration/

**Review by jonathanmetzman — APPROVED** [review]
LGTM

**Pull request merged**